azkaban-executor/azkaban-emr-jobtype/src/main/java/uk/gov/dwp/dataworks/azkaban/jobtype/EMRStep.java

```diff
@@ -204,17 +211,21 @@ private void printLogs(GetLogEventsResult logResult) {
     }
   }
 
-  private boolean isStepCompleted(AmazonElasticMapReduce emr, String clusterId, String stepId) {
+  private Pair<Boolean, String> isStepCompleted(AmazonElasticMapReduce emr, String clusterId, String stepId) {
```

Comment:
This is no longer a predicate function as it returns a Pair instead of boolean, so the name of the function should be something like `getStepStatus`.